router/core/operation_metrics.go

```diff
@@ -0,0 +1,105 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/wundergraph/cosmo/router/internal/metric"
+	"github.com/wundergraph/cosmo/router/internal/otel"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type OperationProtocol int
+
+const (
+	OperationProtocolHTTP OperationProtocol = iota + 1
```

Comment:
Why not use just string constants if we never benefit from this structure?

Reply:
They have a String() method which is what is written to the traces, but I agree with you. Let's cut the middleman and make them strings directly.

Reply:
Changed